From 949b0a2715ff87ff18ec9374dabd1d80a0371f8a Mon Sep 17 00:00:00 2001
From: Douglas Anderson <dianders@chromium.org>
Date: Wed, 16 Jan 2019 11:22:43 -0800
Subject: [PATCH] FIXUP: FROMLIST: arm64: dts: qcom: sdm845: add UFS controller
 FIXUP: FROMLIST: arm64: dts: qcom: sdm845: Add UFS nodes for sdm845-mtp
 FIXUP: CHROMIUM: arm64: dts: qcom: sdm845: Add UFS to Cheza

===
This is a revert of commit df93117e3abb ("FROMLIST: arm64: dts: qcom:
sdm845: add UFS controller") squashed together with the newest version
that's been posted upstream.  This version has landed in the
maintainer's tree as commit cc16687fbd74 ("arm64: dts: qcom: sdm845:
add UFS controller"), but I'm still referencing FROMLIST here because
the for-next branch of the Qualcomm tree is known not to have stable
git hashes.

I've also squashed together a revert of commit 121a1d526d74
("FROMLIST: arm64: dts: qcom: sdm845: Add UFS nodes for sdm845-mtp")
together with the newest version.  That's commit b010fdb4ea58 ("arm64:
dts: qcom: sdm845: Add UFS nodes for sdm845-mtp") in the for-next
branch.  I squashed this in to avoid breaking bisection.

...and finally I've added in a fixup to cheza to match MTP.  Again,
this is squashed to avoid breaking bisection.  When sending cheza
upstream we won't maintain history anyway.

*** NOTE NOTE NOTE: The upstream commit actually contains:
  iommus = <&apps_smmu 0x100 0xf>;

I _didn't_ take that because it will break us until Bjorn's solution
to do 'dma-ranges' is done.  ...so that'll be in (yet another)
future fixup.
===

Add the UFS controller and PHY to SDM845.

Enable the UFS host controller and PHY on sdm845-mtp.

This change enables the UFS controller in Cheza.

Signed-off-by: Evan Green <evgreen@chromium.org>
Signed-off-by: Can Guo <cang@codeaurora.org>
Signed-off-by: Douglas Anderson <dianders@chromium.org>
(am from https://patchwork.kernel.org/patch/10722299/)
(also found at https://lkml.kernel.org/r/20181210192826.241350-4-evgreen@chromium.org)
(am from https://patchwork.kernel.org/patch/10722297/)
(also found at https://lkml.kernel.org/r/20181210192826.241350-5-evgreen@chromium.org)

BUG=b:74254727
TEST=Build and boot; UFS works

Change-Id: I99a4378ae002a64cfb3af21fa52a12b7351d7e48
Signed-off-by: Douglas Anderson <dianders@chromium.org>
Reviewed-on: https://chromium-review.googlesource.com/1415598
Reviewed-by: Evan Green <evgreen@chromium.org>
---
 arch/arm64/boot/dts/qcom/sdm845-cheza.dtsi |  4 ++--
 arch/arm64/boot/dts/qcom/sdm845-mtp.dts    |  4 ++--
 arch/arm64/boot/dts/qcom/sdm845.dtsi       | 11 ++++-------
 3 files changed, 8 insertions(+), 11 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/sdm845-cheza.dtsi b/arch/arm64/boot/dts/qcom/sdm845-cheza.dtsi
index ebb2c496f828..eb0cf3bb2565 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-cheza.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845-cheza.dtsi
@@ -676,7 +676,7 @@ ap_ts_i2c: &i2c14 {
 	status = "okay";
 };
 
-&ufshc1 {
+&ufs_mem_hc {
 	status = "okay";
 	pinctrl-names = "init", "default";
 	pinctrl-0 = <&ufs_dev_reset_assert>;
@@ -686,7 +686,7 @@ ap_ts_i2c: &i2c14 {
 	vcc-max-microamp = <600000>;
 };
 
-&ufsphy1 {
+&ufs_mem_phy {
 	status = "okay";
 
 	vdda-phy-supply = <&vdda_ufs1_core>;
diff --git a/arch/arm64/boot/dts/qcom/sdm845-mtp.dts b/arch/arm64/boot/dts/qcom/sdm845-mtp.dts
index 92eb8a90fd18..af8c6a2445a2 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-mtp.dts
+++ b/arch/arm64/boot/dts/qcom/sdm845-mtp.dts
@@ -377,14 +377,14 @@
 	status = "okay";
 };
 
-&ufshc1 {
+&ufs_mem_hc {
 	status = "okay";
 
 	vcc-supply = <&vreg_l20a_2p95>;
 	vcc-max-microamp = <600000>;
 };
 
-&ufsphy1 {
+&ufs_mem_phy {
 	status = "okay";
 
 	vdda-phy-supply = <&vdda_ufs1_core>;
diff --git a/arch/arm64/boot/dts/qcom/sdm845.dtsi b/arch/arm64/boot/dts/qcom/sdm845.dtsi
index 71964ce79ed5..e24298f98eaf 100644
--- a/arch/arm64/boot/dts/qcom/sdm845.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845.dtsi
@@ -1016,12 +1016,12 @@
 			};
 		};
 
-		ufshc1: ufshc@1d84000 {
+		ufs_mem_hc: ufshc@1d84000 {
 			compatible = "qcom,sdm845-ufshc", "qcom,ufshc",
 				     "jedec,ufs-2.0";
 			reg = <0x1d84000 0x2500>;
 			interrupts = <GIC_SPI 265 IRQ_TYPE_LEVEL_HIGH>;
-			phys = <&ufsphy1_lanes>;
+			phys = <&ufs_mem_phy_lanes>;
 			phy-names = "ufsphy";
 			lanes-per-direction = <2>;
 			power-domains = <&gcc UFS_PHY_GDSC>;
@@ -1054,13 +1054,10 @@
 				<0 0>,
 				<0 0>;
 
-			resets = <&gcc GCC_UFS_PHY_BCR>;
-			reset-names = "rst";
-
 			status = "disabled";
 		};
 
-		ufsphy1: phy@1d87000 {
+		ufs_mem_phy: phy@1d87000 {
 			compatible = "qcom,sdm845-qmp-ufs-phy";
 			reg = <0x1d87000 0x18c>;
 			#address-cells = <1>;
@@ -1073,7 +1070,7 @@
 
 			status = "disabled";
 
-			ufsphy1_lanes: lanes@1d87400 {
+			ufs_mem_phy_lanes: lanes@1d87400 {
 				reg = <0x1d87400 0x108>,
 				      <0x1d87600 0x1e0>,
 				      <0x1d87c00 0x1dc>,
-- 
2.23.0.rc1.153.gdeed80330f-goog

